commit 51c18bf7fdbe3c0546e86cb0fa5866fee07c5261
Author: Chris Wilson <chris@chris-wilson.co.uk>
Date:   Sat Jun 9 12:10:58 2018 +0100

    drm/i915: Squash GEM load failure message (again)
    
    Due to a silent conflict (silent because we are trying to fix the CI
    test that is meant to exercising these failures!) between commit
    51e645b6652c ("drm/i915: Mark the GPU as wedged without error on fault
    injection") and commit 8571a05a9dd0 ("drm/i915: Use GEM suspend when
    aborting initialisation"), we failed to actually squash the error
    message after injecting the load failure.
    
    Rearrange the code to export i915_load_failure() for better logging of
    real errors (and quiet logging of injected errors).
    
    Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
    Cc: Michał Winiarski <michal.winiarski@intel.com>
    Reviewed-by: Michał Winiarski <michal.winiarski@intel.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/20180609111058.2660-1-chris@chris-wilson.co.uk

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index 4187e0688e50..0f28b5ddfd15 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -5491,8 +5491,6 @@ int i915_gem_init(struct drm_i915_private *dev_priv)
 	}
 
 	if (i915_inject_load_failure()) {
-		DRM_DEBUG_DRIVER("Marking the driver as wedged\n");
-		i915_gem_set_wedged(dev_priv);  /* Fail silently! */
 		ret = -EIO;
 		goto err_init_hw;
 	}
@@ -5543,7 +5541,8 @@ int i915_gem_init(struct drm_i915_private *dev_priv)
 		 * for all other failure, such as an allocation failure, bail.
 		 */
 		if (!i915_terminally_wedged(&dev_priv->gpu_error)) {
-			DRM_ERROR("Failed to initialize GPU, declaring it wedged\n");
+			i915_load_error(dev_priv,
+					"Failed to initialize GPU, declaring it wedged!\n");
 			i915_gem_set_wedged(dev_priv);
 		}
 		ret = 0;